bin/rollup/src/network.rs

```diff
@@ -90,19 +90,22 @@ where
         // Create the scroll network manager.
         let scroll_network_manager = ScrollNetworkManager::from_parts(handle.clone(), events);
 
-        // Spawn the scroll network manager.
-        let consensus = PoAConsensus::new(vec![]);
-        let payload_provider = NoopExecutionPayloadProvider;
-
+        // Spawn the engine driver.
         let auth_port = ctx.config().rpc.auth_port;
         let auth_secret = ctx.config().rpc.auth_jwt_secret(ctx.config().datadir().jwt())?;
 
         let engine_api = ScrollAuthEngineApiProvider::new(
             auth_secret,
             self.config.engine_api_url.unwrap_or(format!("http://localhost:{auth_port}").parse()?),
         );
-        let fcs =
-            ForkchoiceState::head_from_genesis(ctx.config().chain.genesis_header().hash_slow());
+        let payload_provider = NoopExecutionPayloadProvider;
+
+        let chain = ctx.config().chain.clone();
+        let fcs = if let Some(named) = chain.chain.named() {
```

Comment:
`NamedChain` is `Copy` so I don't think we need the clone here.
```suggestion
        let fcs = if let Some(named) = ctx.config().chain.chain.named() {
```

Reply:
nice catch, didn't notice if was `Copy`.